### ISSUE TITLE
FR: add some validation to ci-release `iojs+release`

### ISSUE THREAD

**Issue by refack** [nodejs/build#1592]
The current job takes several free-form strings parameters that end up in the build output. Downstream tools assume a consistent format for these value, but the job does not validate those assumptions on the input.

As a result we can create https://nodejs.org/download/nightly/v11.0.0-nightly201806050300f7c68cffa5de7f0099e3afa95093b0bbbe3c/
Or
https://nodejs.org/download/nightly/v12.0.0-nightly20181122.2.94a2c9c8e5/
which will probably bork downstream tools.

tl;dr and some validate code to the job.

Refs: https://github.com/nodejs/node/pull/24551#issuecomment-441144022
